assets/pages/paper-subscription-landing/paperSubscriptionLandingPageReducer.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+// ----- Imports ----- //
+
+import type { CommonState } from 'helpers/page/commonReducer';
+import { type PaperBillingPeriod } from 'helpers/subscriptions';
+import { getQueryParameter } from 'helpers/url';
+import { productPagePeriodFormReducerFor, type State as FormState } from 'components/productPage/productPagePeriodForm/productPagePeriodFormReducer';
+
+export type State = {
+  common: CommonState,
+  page: FormState<PaperBillingPeriod>,
+};
+
+
+// ----- Reducer ----- //
+
+const promoInUrl = getQueryParameter('promo');
```

Comment:
Is it possible to pass this as a param to a higher-order reducer? That way you can avoid running side effects on import of this file.